### PR TITLE
fix flakiness in mv3 multiple restart test

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## Explanation

Fixes flakiness on test `Should continue to support send ETH dApp interactions after service worker re-starts multiple times` as measured by 10 consecutive runs of the test with no errors.

### CODE REVIEW

File: test/e2e/mv3/multiple-restarts.spec.js [hunk 1]
```diff
@@ -257,11 +258,18 @@ describe('MV3 - Restart service worker multiple times', function () {
...
+        await driver.getWindowTitles();
+
         await clickSendButton(driver);
         await driver.waitUntilXWindowHandles(2);
 
         await switchToWindow(driver, WINDOW_TITLES.TestDApp);
         await terminateServiceWorker(driver);
+
+        await driver.getAllWindowHandles();
+        await driver.getWindowTitles();
```
Comment (DDDDDanica) [inline test/e2e/mv3/multiple-restarts.spec.js:271]: Can u explain a bit on these 2 lines ?
For example, I can see inside `waitUntilXWindowHandles`, we have duplicated calls  
```
    while (timeElapsed <= timeout) {
      windowHandles = await this.driver.getAllWindowHandles();
```

File: test/e2e/mv3/multiple-restarts.spec.js [hunk 2]
```diff
@@ -318,14 +326,15 @@ describe('MV3 - Restart service worker multiple times', function () {
     }
 
     async function assertNumberOfTransactionsInPopUp(driver, number) {
+      await sleepSeconds(regularDelayMs);
```
Comment (DDDDDanica) [inline test/e2e/mv3/multiple-restarts.spec.js:329]: can we use `driver.delay` here?

File: test/e2e/webdriver/driver.js [hunk 3]
```diff
@@ -375,11 +375,12 @@ class Driver {
     return await this.driver.getAllWindowHandles();
   }
 
-  async waitUntilXWindowHandles(x, delayStep = 1000, timeout = 5000) {
+  async waitUntilXWindowHandles(x, delayStep = 1000, timeout = this.timeout) {
```
Comment (DDDDDanica) [inline test/e2e/webdriver/driver.js:378]: 💯

File: test/e2e/mv3/multiple-restarts.spec.js [hunk 4]
```diff
@@ -257,11 +258,18 @@ describe('MV3 - Restart service worker multiple times', function () {
...
+        await driver.getAllWindowHandles();
+        await driver.getWindowTitles();
+
         await clickSendButton(driver);
         await driver.waitUntilXWindowHandles(2);
 
         await switchToWindow(driver, WINDOW_TITLES.TestDApp);
         await terminateServiceWorker(driver);
+
+        await driver.getAllWindowHandles();
```
Comment (DDDDDanica) [inline test/e2e/mv3/multiple-restarts.spec.js:270]: Can u explain a bit on this one line ?
For example, I can see inside `waitUntilXWindowHandles`, we have duplicated calls  
```
    while (timeElapsed <= timeout) {
      windowHandles = await this.driver.getAllWindowHandles();
```

File: test/e2e/mv3/multiple-restarts.spec.js (hunk 2 above)
Comment (danjm) [reply test/e2e/mv3/multiple-restarts.spec.js:329]: Better yet, can we just await the creation of the window with the appropriate title?

**Review by brad-decker — DISMISSED** [review]

File: test/e2e/mv3/multiple-restarts.spec.js (hunk 4 above)
Comment (pedronfigueiredo) [reply test/e2e/mv3/multiple-restarts.spec.js]: removed

File: command_output.txt [hunk 5]
```diff
@@ -0,0 +1,72 @@
+Attempt 1 at 2023-07-10 17:06:21:
```
Comment (brad-decker) [inline command_output.txt:1]: I think this file should be removed?

**Review by brad-decker — DISMISSED** [review]

File: command_output.txt (hunk 5 above)
Comment (pedronfigueiredo) [reply command_output.txt]: good catch Brad, I've removed it now

File: test/e2e/webdriver/driver.js [hunk 6]
```diff
@@ -411,6 +413,25 @@ class Driver {
     throw new Error(`No window with title: ${title}`);
   }
 
+  async getWindowTitles(delayStep = 1000, timeout = this.timeout) {
```
Comment (DDDDDanica) [inline test/e2e/webdriver/driver.js:416]: Is this util func used anywhere? If not we can remove it ~

**Review by DDDDDanica — APPROVED** [review]

**Comment by DDDDDanica** [comment]
LGTM !

**Review by brad-decker — APPROVED** [review]

**Pull request merged**